src/lib/components/Badge/Badge.test.tsx

```diff
@@ -0,0 +1,50 @@
+import { act, fireEvent, render } from "@/tests";
+import { Badge } from ".";
+
+describe("Badge", () => {
+  const icon = "https://some-invalid-url/img.jpg";
+
+  test("Should render with required properties", () => {
+    const { getByText } = render(<Badge label="label" />);
+    expect(getByText("label")).toBeInTheDocument();
+  });
+
+  test("Should display icon when valid icon url passed", async () => {
+    const { getByRole, getByText } = render(
+      <Badge label="label" icon={icon} />,
+    );
+
+    const img = getByRole("img", { hidden: true });
+    act(() => {
+      fireEvent.load(img);
+    });
+    expect(img).toBeInTheDocument();
+    expect(img).toHaveAttribute("src", icon);
+    expect(getByText("label")).toBeInTheDocument();
+  });
+
+  test("Should display initials when invalid image url passed", async () => {
```

Comment:
title: "Should display label only..."
And when the URL is incorrect, we still have a padding 
![image](https://github.com/daniknewgarden/exo-ui/assets/38109617/0801a104-3fd1-47cc-b303-60ae7904ecff)